### PR TITLE
Refactor classes

### PR DESCRIPTION
**Pull request by ALEXJAZZ008008**
Refactored and added point cloud output (raw)

### CODE REVIEW

File: src/KinectFrontend/KinectFrontend.h [hunk 1]
```diff
@@ -43,9 +42,57 @@ class KinectFrontend : public QDialog
...
+
+        return 1;
+    }
+
+    inline KinectInterface * get_kinect_interface_ptr()
+    {
+        return m_kinect_interface_ptr;
+    }
+
+    inline int ser_kinect_interface_ptr(KinectInterface * kinect_interface_ptr)
```
Comment (NikEfth) [inline src/KinectFrontend/KinectFrontend.h]: typo

File: src/KinectInterface/KinectBackend/KinectBackend.cpp [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+
+void KinectBackend::depth_callback(freenect_device *m_freenect_device_ptr, void *data, unsigned int timestamp)
+{
+    unsigned short *depth_ptr = (unsigned short *)data;
+
+    for(unsigned short i = 0; i < KinectBackend::getInstance().m_kinect_object_ptr->get_resolution().at(1); ++i)
+    {
+        for(unsigned short j = 0; j < KinectBackend::getInstance().m_kinect_object_ptr->get_resolution().at(0); ++j)
+        {
+            KinectBackend::getInstance().m_kinect_object_ptr->get_depth().at(j).at(i) =
```
Comment (NikEfth) [inline src/KinectInterface/KinectBackend/KinectBackend.cpp]: Could you change this to be a 1D array (or vector)?

File: src/KinectInterface/KinectBackend/KinectBackend.cpp [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+    }
+
+    m_message += "Connection terminated!!\n";
+
+    return 1;
+}
+
+void KinectBackend::depth_callback(freenect_device *m_freenect_device_ptr, void *data, unsigned int timestamp)
+{
+    unsigned short *depth_ptr = (unsigned short *)data;
```
Comment (NikEfth) [inline src/KinectInterface/KinectBackend/KinectBackend.cpp]: static_cast

File: src/KinectInterface/KinectBackend/KinectBackend.cpp [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+
+        m_current_tilt_state_ptr = nullptr;
+    }
+
+    m_message += "Connection terminated!!\n";
+
+    return 1;
+}
+
+void KinectBackend::depth_callback(freenect_device *m_freenect_device_ptr, void *data, unsigned int timestamp)
```
Comment (NikEfth) [inline src/KinectInterface/KinectBackend/KinectBackend.cpp]: check if empty? or the same?

File: src/KinectInterface/KinectBackend/KinectBackend.cpp [hunk 5]
```diff
@@ -0,0 +1,410 @@
...
+            KinectBackend::getInstance().m_kinect_object_ptr->get_depth().at(j).at(i) =
+                    depth_ptr[(KinectBackend::getInstance().m_kinect_object_ptr->get_resolution().at(0) * i) + j];
+        }
+    }
+
+    KinectBackend::getInstance().m_kinect_object_ptr->set_timestamp(timestamp);
+
+    KinectBackend::getInstance().m_kinect_object_ptr->set_got_depth(true);
+
+    KinectBackend::getInstance().append_depth_message("Received depth frame at " + to_string(timestamp) + "\n");
```
Comment (NikEfth) [inline src/KinectInterface/KinectBackend/KinectBackend.cpp]: Reduce the number of characters.

Comment (NikEfth) [reply src/KinectInterface/KinectBackend/KinectBackend.cpp]: move at interface

File: src/KinectInterface/KinectBackend/KinectBackend.cpp [hunk 6]
```diff
@@ -0,0 +1,410 @@
...
+            KinectBackend::getInstance().m_kinect_object_ptr->get_video().at(j).at(i).at(2) =
+                    video_ptr[(((KinectBackend::getInstance().m_kinect_object_ptr->get_resolution().at(0) * i) + j) * 3) + 2];
+        }
+    }
+
+    KinectBackend::getInstance().m_kinect_object_ptr->set_timestamp(timestamp);
+
+    KinectBackend::getInstance().m_kinect_object_ptr->set_got_video(true);
+
+    KinectBackend::getInstance().append_video_message("Received video frame at "  + to_string(timestamp) + "\n");
```
Comment (NikEfth) [inline src/KinectInterface/KinectBackend/KinectBackend.cpp]: same as above

File: src/KinectInterface/KinectBackend/KinectBackend.cpp [hunk 7]
```diff
@@ -0,0 +1,410 @@
...
+}
+
+int KinectBackend::calculate_point_cloud()
```
Comment (NikEfth) [inline src/KinectInterface/KinectBackend/KinectBackend.cpp]: move in the point cloud numerics

File: src/KinectInterface/KinectBackend/KinectBackend.cpp [hunk 8]
```diff
@@ -0,0 +1,410 @@
...
+{
+    vector<float>::iterator vector_iterator = m_kinect_object_ptr->get_point_cloud().begin();
+
+    for(unsigned short i = 0; i < m_kinect_object_ptr->get_resolution().at(1); ++i)
+    {
+        for(unsigned short j = 0; j < m_kinect_object_ptr->get_resolution().at(0); ++j)
+        {
+            *vector_iterator = (j - (m_kinect_object_ptr->get_resolution().at(1) / 2.0f)) * (m_kinect_object_ptr->get_depth().at(j).at(i) - 10) * 0.0021f;
```
Comment (NikEfth) [inline src/KinectInterface/KinectBackend/KinectBackend.cpp]: 0.0021f make it a variable. The default value will be set in the constructor. Create a set function

Comment (NikEfth) [inline src/KinectInterface/KinectBackend/KinectBackend.cpp]: m_kinect_object_ptr->get_resolution().at(1) cast to float

File: src/KinectInterface/KinectBackend/KinectBackend.cpp (hunk 7 above)
Comment (NikEfth) [inline src/KinectInterface/KinectBackend/KinectBackend.cpp]: If you find the link add it here!

**Review by NikEfth — CHANGES_REQUESTED** [review]

**Comment by NikEfth** [comment]
I broken the front end into pieces. Now konnector is the class which handles the managerial aspects. Logger prints the log msgs, settings allow you to set option (but are not passed at the backend). 
Your tasked (in addition to those marked here are to implement all the `\todo` I have added in the source code. 
There is commentary to help, but we'll skype tomorrow noon for details.

**Comment by ALEXJAZZ008008** [comment]
Please merge me!

**Pull request merged**